src/filter/hierarchical.rs

```diff
@@ -423,6 +436,26 @@ impl<Key, Filter, Child> HierarchicalFilters<Key, Filter, Child> {
         self.children.into_iter().flatten().collect()
     }
 
+    pub(crate) async fn for_each<F, R>(&mut self, f: F)
+    where
+        F: Fn(Child) -> R + Copy,
+        R: futures::Future<Output = Child>,
+    {
+        self.children = self
+            .children
+            .drain(..)
+            .map(|o| async move {
+                if let Some(c) = o {
+                    Some(c.map_async(f).await)
+                } else {
+                    None
+                }
+            })
+            .collect::<FuturesUnordered<_>>()
+            .collect::<Vec<_>>()
```

Comment:
The only place, where `for_each` is used just ignores the result of this function. So it is better to not collect results into a Vector

Reply:
This is needed to recreate source vector after parallel processing

Reply:
Function removed